server/utils/tans.py

```diff
@@ -83,9 +92,23 @@ def uniques(n: int, length: int = 5) -> list[Tan]:
     """
     if n > possible_tans(length):
         raise ValueError(
-            f"Cannot generate {n} unique TANs of length {length}. Maximum possible TANs is {possible_tans(length)}."
+            f"Cannot generate {n} unique TANs of length {length}. Maximum possible TANs of this length are "
+            f"{possible_tans(length)}."
         )
+    # Retrieve existing tans
     tans = set()
-    while len(tans) < n:
+    with create_app(csrf, db).app_context():
+        db_tans = [tan for tan, in db.session.query(models.Player.tan).all()]
+        tans.update(db_tans)
+    existing_tans = set(tans)
```

Comment:
why again the set constructor? Doesnt the update command creates a new set of both. Further the db_tans can be assumed as set and due to its primary characteristic.

Reply:
1. To create a deep copy of `tans`, otherwise I would have to variables pointing to the same object
2. `update()` is a way to union a set with an iterable, so to add all member of a list to a set
3. Correct, what's your point?